azure-kusto-ingest/azure/kusto/ingest/_descriptors.py

```diff
@@ -38,24 +32,12 @@ def __init__(self, path, size=0, source_id=None):
         assert_uuid4(source_id, "source_id must be a valid uuid4")
         self.source_id = source_id
         self.stream_name = os.path.basename(self.path)
-        if self.path.endswith(".gz") or self.path.endswith(".zip"):
-            self.zipped_stream = open(self.path, "rb")
-            if not self.size or self.size <= 0:
+        if not self.size or self.size <= 0:
+            if self.path.endswith(".gz") or self.path.endswith(".zip"):
                 # TODO: this can be improved by reading last 4 bytes
                 self.size = int(os.path.getsize(self.path)) * 5
-        else:
-            self.size = int(os.path.getsize(self.path))
-            self.stream_name += ".gz"
-            self.zipped_stream = BytesIO()
-            with open(self.path, "rb") as f_in, GzipFile(filename="data", fileobj=self.zipped_stream, mode="wb") as f_out:
-                shutil.copyfileobj(f_in, f_out)
-            self.zipped_stream.seek(0)
-
-    def delete_files(self):
-        """Deletes the gz file if the original file was not zipped.
-        In case of success deletes the original file as well."""
-        if self.zipped_stream is not None:
-            self.zipped_stream.close()
```

Comment:
why?

Reply:
I don't keep the stream inside the descriptor anymore - therefore it's not its responsibility to close it.